tests/MongoDB.Driver.Tests/AuthenticationTests.cs

```diff
@@ -132,7 +134,7 @@ public void Authentication_succeeds_when_user_has_Scram_Sha_1_Mechanism_and_mech
             settings.Credential = MongoCredential
                 .FromComponents(mechanism: null, source: null, username: userName, password: password);
 
-            AssertAuthenticationSucceeds(settings);
+            AssertAuthenticationSucceeds(settings, shouldSpeculativelyAuthenticateIfPossible: false);
```

Comment:
We speculatively authenticate with scramsha256 when the authmechanism isn't specified
https://github.com/mongodb/specifications/blob/master/source/mongodb-handshake/handshake.rst#speculative-authentication